test/bootstrap_form_test.rb

```diff
@@ -12,6 +12,11 @@ class BootstrapFormTest < ActionView::TestCase
       </form>
     HTML
     assert_equivalent_xml expected, bootstrap_form_for(@user) { |f| nil }
+    if Gem::Version.new(::Rails::VERSION::STRING).release >= Gem::Version.new('5.1.0')
+      # TODO: Could argue that we should test without `local: true`
+      assert_equivalent_xml remove_default_ids_for_rails_5_1(expected.gsub(/ class="new_user" id="new_user"/, "")),
+                            bootstrap_form_with(model: @user, local: true) { |f| nil }
```

Comment:
I think this is really the only test we need for this PR. But I would split it into 2 tests: one for Rails 5.1, and a second for Rails 5.2+. I prefer it be written as two separate, *new* tests (i.e. not bundled under `test "default-style forms"`). And they should each have their own expected HTML rather than mutating a single expectation with `remove_default_ids_for_rails_5_1`. I believe repeating the HTML makes it much more explicit as to what is being tested and what is expected, which is very important value for tests.

Reply:
Yes, I'll do that. (And see my next comment.)